src/components/tabs/__internal__/tab-title/tab-title.component.tsx

```diff
@@ -0,0 +1,295 @@
+import React, { useRef, useState } from "react";
+import {
+  StyledTabTitleButton,
+  StyledTabTitleLink,
+  StyledTitleContent,
+  StyledLayoutWrapper,
+  StyledSelectedIndicator,
+} from "./tab-title.style";
+import tagComponent from "../../../../__internal__/utils/helpers/tags/tags";
+import ValidationIcon from "../../../../__internal__/validations/validation-icon.component";
+import Icon from "../../../icon";
+import Events from "../../../../__internal__/utils/helpers/events";
+import { TooltipProvider } from "../../../../__internal__/tooltip-provider";
+
+export interface TabTitleContextProps {
+  isInTab?: boolean;
+}
+
+export interface TabTitleProps {
+  /** Identifier used for testing purposes */
+  "data-role"?: string;
+  title?: string;
+  id?: string;
+  dataTabId?: string;
+  className?: string;
+  children?: React.ReactNode;
+  isTabSelected?: boolean;
+  position?: "top" | "left";
+  errorMessage?: string;
+  warningMessage?: string;
+  infoMessage?: string;
+  error?: boolean;
+  warning?: boolean;
+  info?: boolean;
+  borders?: boolean;
+  noLeftBorder?: boolean;
+  noRightBorder?: boolean;
+  alternateStyling?: boolean;
+  isInSidebar?: boolean;
+  siblings?: React.ReactNode;
+  titlePosition?: "before" | "after";
+  href?: string;
+  tabIndex?: string;
+  size?: "default" | "large";
+  align?: "left" | "right";
+  customLayout?: React.ReactNode;
+  onClick: (
+    ev: React.MouseEvent<HTMLButtonElement | HTMLAnchorElement>
+  ) => void;
+  onKeyDown: (
+    ev: React.KeyboardEvent<HTMLButtonElement | HTMLAnchorElement>
+  ) => void;
+}
+
+export const TabTitleContext = React.createContext({});
+
+const TabTitle = React.forwardRef(
+  (
+    {
+      isTabSelected,
+      dataTabId,
+      "data-role": dataRole,
+      title,
+      position,
+      error,
+      warning,
+      info,
+      size = "default",
+      onClick,
+      titlePosition = "before",
+      siblings,
+      errorMessage = "",
+      warningMessage = "",
+      infoMessage = "",
+      borders,
+      alternateStyling = false,
+      noLeftBorder = false,
+      noRightBorder = false,
+      customLayout,
+      isInSidebar,
+      href,
+      onKeyDown,
+      align,
+      tabIndex,
+      ...tabTitleProps
+    }: TabTitleProps,
+    ref: React.ForwardedRef<HTMLElement>
+  ) => {
+    const isHref = !!href;
+    const hasAlternateStyling = alternateStyling || isInSidebar;
+    const hasFailedValidation = error || warning || info;
+    const [shouldShowTooltip, setShouldShowTooltip] = useState(false);
+    const hasHover = useRef(false);
+
+    const showTooltip = () => {
+      setShouldShowTooltip(true);
+    };
+
+    const hideTooltip = () => {
+      // we only provide objects as a ref to this internal component so the type assertion is safe
+      if (
+        (ref as React.RefObject<HTMLElement>)?.current !==
```

Comment:
Good point, I generally avoid type assertions if it's at all possible. I think I did it this way because of what I explained in the comment - it's completely safe here and therefore the check is unncessary. But we don't know that will always be the case so I'm happy to change it.